File: app/src/main/java/com/mirash/familiar/tool/LabelString.java
package com.mirash.familiar.tool;

import android.text.TextUtils;

import androidx.annotation.NonNull;

/**
 * @author Mirash
 */
class LabelString implements CharSequence {
    private final String str;

    public LabelString(String label, String value) {
        str = TextUtils.isEmpty(value) ? "" : label + ": " + value;
    }

    @Override
    public int length() {
        return str.length();
    }

    @Override
    public char charAt(int i) {
        return str.charAt(i);
    }

    @NonNull
    @Override
    public CharSequence subSequence(int i, int i1) {
        return str.subSequence(i, i1);
    }

    @NonNull
    @Override
    public String toString() {
        return str;
    }
}

File: app/src/main/java/com/mirash/familiar/tool/LabelString.kt
package com.mirash.familiar.tool

/**
 * @author Mirash
 */
class LabelString(label: String, value: String?) : CharSequence {
    private val str: String = if (value.isNullOrEmpty()) "" else "$label: $value"
    override val length: Int
        get() = str.length

    override fun get(index: Int): Char = str[index]

    override fun subSequence(startIndex: Int, endIndex: Int): CharSequence = str.subSequence(startIndex, endIndex)

    override fun toString(): String = str
}
